### PR TITLE
APPDEV-1196 App activity monitoring notification disappears.

### PR DESCRIPTION
**Pull request by Sivateja-CMI**
-- This is happening when device goes to doze mode. To fix this, added device_awake alarm broadcast for starting foreground service. This fires only after device goes into doze mode and OS above OREO.

### CODE REVIEW

File: app/src/main/java/nu/yona/app/api/receiver/YonaReceiver.java [hunk 1]
```diff
@@ -24,46 +27,112 @@
...
 				break;
 			case Intent.ACTION_SCREEN_OFF:
-				Logger.loge("SEND_Screen Off", "Screen Off");
-				AppUtils.setNullScheduler();
-				AppUtils.sendLogToServer(AppConstant.ONE_SECOND);
-				AppUtils.stopService(context);
+				handleScreenOffBroadcast(context);
+				break;
+			case AppConstant.WAKE_UP:
+				handleWakeUpAlarmInOroeAndAbove(context);
```
[the remark this answers is not in the document]
Comment (Sivateja-CMI) [reply app/src/main/java/nu/yona/app/api/receiver/YonaReceiver.java]: Done.

File: app/src/main/java/nu/yona/app/utils/AppConstant.java [hunk 2]
```diff
@@ -72,6 +72,8 @@
 	int ONE_SECOND = 1000;
 
 	int THREE_SECOND = 3000;
+
+	int TEN_SECONDS = 10000;
```
[the remark this answers is not in the document]
Comment (Sivateja-CMI) [reply app/src/main/java/nu/yona/app/utils/AppConstant.java]: Updated code and kept constant ONE_SECOND, as this is used across application.

File: app/src/main/java/nu/yona/app/api/receiver/YonaReceiver.java [hunk 3]
```diff
@@ -24,46 +27,112 @@
...
+			scheduleNextAlarmToCheckIfDeviceIsInteractive(context, AppConstant.TEN_SECONDS);
+		}
+	}
+
+	@TargetApi(Build.VERSION_CODES.O)
+	private void handleWakeUpAlarmInOroeAndAbove(Context context)
+	{
+		// Device is awake from doze/sleep (it can be because of user interaction or of some silent Push notifications).
+		// We should start service only when device is interactive else schedule next alarm
+		Logger.logi("WAKE", "Alarm Fired");
```
[the remark this answers is not in the document]
Comment (Sivateja-CMI) [reply app/src/main/java/nu/yona/app/api/receiver/YonaReceiver.java]: Removed.

File: app/src/main/java/nu/yona/app/api/receiver/YonaReceiver.java [hunk 4]
```diff
@@ -24,46 +27,112 @@
...
+		if (isDeviceInteractive(context))
+		{
+			Logger.logi("WAKE", "Device interactive. Service started");
```
[the remark this answers is not in the document]
Comment (Sivateja-CMI) [reply app/src/main/java/nu/yona/app/api/receiver/YonaReceiver.java]: Removed. These logs are not needed and I had the idea to remove once this is tested in your Galaxy device as well.

File: app/src/main/java/nu/yona/app/api/receiver/YonaReceiver.java [hunk 5]
```diff
@@ -24,46 +27,112 @@
...
+			startService(context);
+			AppUtils.startVPN(context, false);
+		}
+		else
+		{
+			Logger.logi("WAKE", "Scheduled Next Alarm");
```
[the remark this answers is not in the document]
Comment (Sivateja-CMI) [reply app/src/main/java/nu/yona/app/api/receiver/YonaReceiver.java]: Removed. These logs are not needed and I had the idea to remove once this is tested in your Galaxy device as well.

File: app/src/main/java/nu/yona/app/api/receiver/YonaReceiver.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (Bert-R) [reply app/src/main/java/nu/yona/app/api/receiver/YonaReceiver.java]: You say it's removed, but I still see it.

**Pull request merged**